modules/tensor_mechanics/include/actions/DomainIntegralAction.h

```diff
@@ -49,6 +49,9 @@ class DomainIntegralAction : public Action
   std::set<INTEGRAL> _integrals;
   const std::vector<BoundaryName> & _boundary_names;
   std::vector<Point> _crack_front_points;
+  UserObjectName _crack_front_points_provider;
+  unsigned int _number_crack_front_points;
```

Comment:
Just one tiny thing: it would better if this were named `_num_points_from_provider` as it is in CrackFrontDefinition. You could actually just eliminate this as a member variable in this class, and grab it from the parameters when you need it.

Reply:
Yes, I will make the change when I fix the closed_loop issue. Thanks.